### PR TITLE
docs: update readme

### LINKED ISSUE #4665: docs: update README to include clear project value proposition
I'm having trouble figuring out the value proposition of this project after reading the README & looking at the project website.

Here's the current description:

> Ibis is a Python library to help you write expressive analytics at any scale, small to large. Its goal is to simplify analytical workflows and make you more productive.

Looks like you can create an Ibis table per the [pandas Documentation page](https://ibis-project.org/docs/3.2.0/backends/Pandas/).  But why should I create an Ibis table?  What value does this provide me?

### PR DESCRIPTION
**Pull request by p-a-a-a-trick**
Resolves #4665.

### CODE REVIEW

**Comment by p-a-a-a-trick** [comment]
@marlenezw

**Comment by cpcloud** [comment]
@p-a-a-a-trick @marlenezw I pushed up a few commits, let me know whether you like them. The primary change is to use level-3 headings instead of a bulleted list.

File: README.md [hunk 1]
```diff
@@ -1,31 +1,56 @@
...
-```sh
-conda install ibis-framework -c conda-forge
-```
+## Why Use Ibis?
+
+Ibis aims to be a future-proof solution to interacting with data using Python and can accomplish this goal through its main features:
+
+- **Familiar API**: Ibis’s API design borrows from popular APIs like pandas and dplyr that most users already know and like to use.
+- **Consistent syntax**: After learning how to use Ibis, users can use the same API and concepts no matter where their data lives.
+  **Learn once, write anywhere**.
```
Comment (datametrician) [inline README.md:30]: I guess we're going with this. I'm growing more fond of this but I still think "learn" is a bit passive. I'd like to push us to think about something more aggressive without alienating users. Maybe:
*A Universal Python API
*Standardized Python Query Language
*Python made Portable
Again, feel free to over rule me, but learn in my head highlights the lack of guarantees, when in reality there are clusters of functionality that are very portable, and those clusters needs to be articulated better.

Comment (cpcloud) [reply README.md:30]: How about `Portable Python Analytics`?

Comment (p-a-a-a-trick) [reply README.md:30]: Yeah I'm not a big fan of "learn" if I'm being honest; if we went with "Learn once, write anywhere" as a bit then the first word someone runs into wrt Ibis will be "learn", which sounds like a homework assignment.

Learning isn't a bad thing obviously, and it's something a new user will be doing anyways, but it, to me, feels like an unwelcoming hurdle to Ibis vs demonstrating that Ibis is something you can jump into and immediately get long-term value out of _while_ learning a universal interface to tabular data.

For this particular bullet, does this make sense?

<hr>

- **Consistent syntax**: After learning how to use Ibis, users can use the same API and concepts no matter where their data lives. **Ibis aims to be universal python API for tabular data, big or small.**.

<hr>

Or do we think there's a nuance that this is eliding?

Comment (cpcloud) [reply README.md:30]: I think you can probably just keep the last sentence.

File: README.md [hunk 2]
```diff
@@ -1,31 +1,55 @@
...
 
-```sh
-conda install ibis-framework -c conda-forge
-```
+## Why Use Ibis?
+
+Ibis aims to be a future-proof solution to interacting with data using Python and can accomplish this goal through its main features:
+
+- **Familiar API**: Ibis’s API design borrows from popular APIs like pandas and dplyr that most users already know and like to use.
+- **Consistent syntax**:  Ibis aims to be universal python API for tabular data, big or small.
```
Comment (p-a-a-a-trick) [inline README.md:29]: @datametrician , @cpcloud , @marlenezw thoughts?

**Comment by cpcloud** [comment]
Frankly, this PR is ready to merge as is IMO.

1. There's nothing incorrect here.
2. It's a huge improvement over the existing README.md

Perfect is the enemy of the good. Ship it.

File: README.md (hunk 1 above)
Comment (saulpw) [reply README.md]: @datametrician said:
> Again, feel free to over rule me, but learn in my head highlights the lack of guarantees, when in reality there are clusters of functionality that are very portable, and those clusters could be articulated better.

We're not necessarily making a long-term decision, just incremental progress to push the ball forward.  It's too easy to get caught up in all these details when reviewing a PR; the ideal is to take this PR and then whenever you/we/whomever has some better words, they can submit a new PR and we'll take that one too.  Even if it's tomorrow :)

**Review by cpcloud — APPROVED** [review]

**Comment by marlenezw** [comment]
Sorry for the lack of comments from me I was traveling back home yesterday! Happy this was merged and will probably add a few more improvements as another PR soon!

File: README.md (hunk 1 above)
Comment (datametrician) [reply README.md]: Noted @saulpw and understood.

I love Portable Python Analytics @cpcloud!

**Pull request merged**